### ISSUE TITLE
schema for resource: Do not allow setting the "id" for regular users

### ISSUE THREAD

**Issue by paulmueller** [ckan/ckan#5285]
### CKAN Version if known (or site URL)
2.8.3 but probably also current master

### Please describe the expected behaviour
It should not be possible to set the resource id when uploading a new reasource via the API as a non-admin.

I would expect something like (as you get when you try to set the id during package_create):
```
    "error": {
        "__type": "Validation Error",
        "id": [
            "The input field id was not expected."
        ]}
```

### Please describe the actual behaviour
The specified id is accepted. I think this is due to the resource schema in
https://github.com/ckan/ckan/blob/07e4dc9080e771f9c5e1fedb5a623bcafbbf632d/ckan/logic/schema.py#L34
which should be more like that of the package
https://github.com/ckan/ckan/blob/07e4dc9080e771f9c5e1fedb5a623bcafbbf632d/ckan/logic/schema.py#L111-L112

### What steps can be taken to reproduce the issue? 

Create Dataset
```
http http://192.168.122.247/api/3/action/package_create name=atestpkg owner_org=atestorg Authorization:XXXX
```

Create Package
```
curl -H'Authorization: XXXX' 'http://192.168.122.247/api/3/action/resource_create' --form upload=@testfile.txt --form package_id=atestpkg --form id=thisshouldnotbepossible
```

**Comment by paulmueller** [comment]
Needless to say that this is problematic when the id provided is less than 6 characters long (Internal Server Error because uploader stores the files in a directory tree that uses the first six characters as folder names).

**Comment by wardi** [comment]
It's useful to assign resource ids when harvesting datasets from one ckan to another, but we should at least be enforcing that the ids are UUIDs, something like https://github.com/open-data/ckanext-canada/blob/a46856aed344c0e162d0b5f49909b498000edaeb/ckanext/canada/validators.py#L100-L110

**Comment by paulmueller** [comment]
But does that apply to regular users (non-admins)? And does it make sense that a regular user can set the UUID of a resource, but *not* for the dataset?

**Comment by wardi** [comment]
you're right I think we should have the same rules for both. For most sites IMO that should be allow users to set the ids but force them to use normalized UUID values.

**Comment by paulmueller** [comment]
I guess you would then also have to add a check for UUID collisions, or is that already implemented somehow?

**Comment by wardi** [comment]
being worked on in #7868

**Comment by paulmueller** [comment]
resources are a difficult topic: https://github.com/ckan/ckan/pull/8069#issuecomment-1954101873
